File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def is_prime(num):
    if num < 2:
        return False
    
    for i in range(2, num):
        if num % i == 0:
            return False
            
    return True
        

def solution(numbers):
    answer = 0
    nums = []
    
    for i in range(1, len(numbers) + 1):
        nums.append(list(set(map("".join, permutations(numbers, i)))))
    per = list(set(map(int, set(sum(nums, [])))))
    
    for i in per:
        if is_prime(i):
            answer += 1
    
    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    boolean[] visited;
    Set<Integer> lst = new HashSet<>();
    
    private boolean isPrime(int n) {
        if (n < 2) return false;
        
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) return false;
        }
        
        return true;
    }
    
    private void dfs(String numbers, String current, int length) {
        // base
        if (current.length() == length) {
            int num = Integer.parseInt(current);
            lst.add(num);
            return;
        }
            
        for (int i = 0; i < numbers.length(); i++) {
            
            if (!visited[i]) {
                visited[i] = true;
                dfs(numbers, current + numbers.charAt(i), length);
                visited[i] = false;
            }
        }
    }
    
    public int solution(String numbers) {
        int answer = 0;
        visited = new boolean[numbers.length()];
        
        for (int i = 0; i < numbers.length(); i++) {
            dfs(numbers, "", i+1);
        }
        
        for (int x : lst) {
            if (isPrime(x)) answer++;
        }
        
        return answer;
    }
}